src/pm/hydra/pm/pmiserv/common.c

```diff
@@ -173,6 +173,9 @@ HYD_status HYD_pmcd_pmi_add_kvs(const char *key, char *val, struct HYD_pmcd_pmi_
         kvs->key_pair = key_pair;
     }
     else {
+#ifdef HAVE_ERROR_CHECKING
```

Comment:
Error checking is on by default, so this doesn't really help.  You need a different macro for this.